modules/sops/default.nix

```diff
@@ -1,18 +1,21 @@
-{ config, lib, pkgs, ... }:
+{ config, options, lib, pkgs, ... }:
 
 with lib;
 
 let
   cfg = config.sops;
+  opts = options.sops;
   users = config.users.users;
   sops-install-secrets = cfg.package;
   sops-install-secrets-check = cfg.validationPackage;
-  regularSecrets = lib.filterAttrs (_: v: !v.neededForUsers) cfg.secrets;
-  secretsForUsers = lib.filterAttrs (_: v: v.neededForUsers) cfg.secrets;
-  secretType = types.submodule ({ config, ... }: {
+  secrets = mapAttrs (_: secret: removeAttrs secret ["sopsFile"]) cfg.secrets;
+  regularSecrets = lib.filterAttrs (_: v: !v.neededForUsers) secrets;
+  secretsForUsers = lib.filterAttrs (_: v: v.neededForUsers) secrets;
+  secretType = types.submodule ({ config, options, ... }: {
     config = {
-      sopsFile = lib.mkOptionDefault cfg.defaultSopsFile;
-      sopsFileHash = mkOptionDefault (optionalString cfg.validateSopsFiles "${builtins.hashFile "sha256" config.sopsFile}");
+      sopsFile = mkOptionDefault cfg.defaultSopsFile;
+      sopsFiles = if options.sopsFile.isDefined then warn "`sops.secrets.<name>.sopsFile` is being deprecated, use `sops.secrets.<name>.sopsFiles` instead" [ config.sopsFile ] else (lib.mkOptionDefault cfg.defaultSopsFiles);
```

Comment:
Can we not add `defaultSopsFile` to the new `defaultSopsFiles` list?

Reply:
Would it not be better to keep `defaultSopsFile` and `defaultSopsFiles` separate and place `sopsFile` before `sopsFiles` per secret?
If you merge in default you would need to remove `sopsFile` 
otherwise `sopsFiles` would become [ `sopsFile` `defaultSopsFile` ] ++ `defaultSopsFiles`.

The only way I was able to implement this feels however a bit hacky

```nix
    config = mkMerge [{
      sopsFile = mkOptionDefault cfg.defaultSopsFile;
      sopsFiles = mkOptionDefault cfg.defaultSopsFiles;
      sopsFilesHash = mkOptionDefault (optionals cfg.validateSopsFiles (forEach config.sopsFiles (builtins.hashFile "sha256")));
    } 
    {
      sopsFiles = mkIf (config.sopsFile != null) (mkOverride options.sopsFile.highestPrio (mkBefore [config.sopsFile]));
    }];
```

The effect of this is that when both `sopsFile` and `sopsFiles` are defined (and not null) with different priorities only one will make it into `config.sopsFiles`